src/compiler/transformers/module/module.ts

```diff
@@ -1502,9 +1502,30 @@ namespace ts {
          * @param value The exported value.
          * @param location The location to use for source maps and comments for the export.
          */
-        function createExportExpression(name: Identifier, value: Expression, location?: TextRange) {
+        function createExportExpression(name: Identifier, value: Expression, location?: TextRange, liveBinding?: boolean) {
             return setTextRange(
-                createAssignment(
+                liveBinding && languageVersion !== ScriptTarget.ES3 ? createCall(
+                    createPropertyAccess(
+                        createIdentifier("Object"),
+                        "defineProperty"
+                    ),
+                    /*typeArguments*/ undefined,
+                    [
+                        createIdentifier("exports"),
+                        createLiteral(name),
+                        createObjectLiteral([
+                            createPropertyAssignment("enumerable", createLiteral(/*value*/ true)),
+                            createPropertyAssignment("get", createArrowFunction(

```

Comment:
You need to use `createFunctionExpression` instead of `createArrowFunction`. The `module` transform is run *after* all other language-specific transforms have been run, so the arrow function won't be correctly down-leveled when targeting ES5.